### ISSUE TITLE
cv::VideoCapture.open(RTSP URL string) crashes on Android and iOS.

### ISSUE THREAD

**Issue by omatrot** [opencv/opencv#11995]
##### System information (version)

- OpenCV => 3.4.1
- Operating System / Platform => Android
- Compiler => Android Studio

##### Detailed description
I'm using OpenCV native C++ code.
The following code:
cv::VideoCature.open("http://192.168.1.107/videostream.asf?user=admin&pwd=&resolution=32");

crashes and show the following variables (here in Android Studio):

cv::param_VIDEOCAPTURE_DEBUG = Cannot find the execution point
Signal = SIGBUS (signal SIGBUS: illegal alignment)
__cv_trace_location_extra_fn118 = {cv::utils::trace::details::Region::LocationExtraData * | 0x0} NULL
__cv_trace_location_fn118 = {const LocationStaticStorage} 
this = {cv::VideoCapture * | 0x8bf893f0} 0x8bf893f0
filename = {const cv::String &} 
__region_fn = {const cv::utils::trace::details::Region} 

Please note that the URL is a RTSP stream. The same code works fine on Linux.

I'm sorry I can't find a way to copy the stack trace in Android Studio.
The last frame is:
cv::String::deallocate() 0x000000008b0829ce

##### Steps to reproduce

cv::VideoCapture cap;
cap.open("http://192.168.1.107/videostream.asf?user=admin&pwd=&resolution=32");

**Comment by omatrot** [comment]
![capture d ecran 2018-07-17 a 15 57 05](https://user-images.githubusercontent.com/4366356/42822816-4d435632-89dc-11e8-885d-93bce35db336.png)

![capture d ecran 2018-07-17 a 16 00 29](https://user-images.githubusercontent.com/4366356/42822832-5374f6c8-89dc-11e8-8907-f1cd8d35a8c3.png)

**Comment by alalek** [comment]
There is no support for RSTP streams in builds for Android platform.

Stack trace shows code from builtin OpenCV MJPEG codec. It is hard to say what is going wrong, but this codec definitely can't handle RSTP streams.

You need to rebuild OpenCV with FFmpeg / GStreamer for Android (they can handle RSTP streams).
Unfortunately, we can't provide complete actual recipes/steps how to do that - you can try to ask for help on Users Q/A forum: http://answers.opencv.org

**Comment by omatrot** [comment]
We've solved the problem by aplying the following flags for compilation:
`-fpack-struct=8 -mllvm -arm-promote-constant=0`

The illegal alignment error lead us to use those options.

**Comment by takacsd** [comment]
Have the same error with a simple mp4 file, but rebuilding opencv with the flags doesn't seem to help...
@omatrot can you explain exactly what was causing the error in your case?

**Comment by omatrot** [comment]
@takacsd We had this problem on a 32 bits armeabi-v7a Android phone.
Low level c++ code and compilation options could be tricky.
If these options are not helping, then you're facing another problem that deserves its own issue.

**Comment by takacsd** [comment]
Yes, it is also a 32 bits armeabi-v7a Android phone, with the same error at the same place.
Anyway I don't use VideoCapture anymore, seems like everything else is working fine.

**Comment by YasserMojahed** [comment]
> We've solved the problem by aplying the following flags for compilation:
> `-fpack-struct=8 -mllvm -arm-promote-constant=0`
> 
> The illegal alignment error lead us to use those options.

Hi Omatrot,
I used to use VideoCapture for RTSP stream in Linux C++
Now, I am using VideoCapture in Java Android and it doesnt work. Open() method returns false.
I am going to Try Native C++ with NDK in AndroidStudio.
so seems u tried the same and that wont work. 
Now I am confused. Where I must add these flags?
-fpack-struct=8 -mllvm -arm-promote-constant=0
Thanks for help
